# QUANTS: Output

## Read trimming

### cutadapt

<details markdown="1">
<summary>Output files</summary>

* `cutadapt/`
    * `*.trim.fastq.gz`: Trimmed sequences (FASTQ).
    * `*.cutadapt.log`: Log file containing trimming statistics.

</details>

[Cutadapt](https://cutadapt.readthedocs.io/en/stable/) finds and removes adapter sequences, primers, poly-A tails and other types of unwanted sequence from your high-throughput sequencing reads. For further reading and documentation see the [cutadapt user guide](https://cutadapt.readthedocs.io/en/stable/guide.html).

## Read merging

### FLASH2

<details markdown="1">
<summary>Output files</summary>

* `flash2/`
    * `*.merged.extendedFrags.fastq.gz`: Merged sequences (FASTQ).
    * `*.merged.notCombined_[1|2].fastq.gz`: Sequences that could not be merged (FASTQ).
    * `*.merged.hist`: Frequency of merged sequence lengths.

</details>

[FLASH](https://github.com/dstreett/FLASH2) (Fast Length Adjustment of SHort reads) is an accurate and fast tool to merge paired-end reads.

## Read filtering

### SeqKit seq

<details markdown="1">
<summary>Output files</summary>

* `seqkit_seq/`
    * `*.filtered.fq.gz`: Filtered sequences (FASTQ).

</details>

[SeqKit](https://bioinf.shenwei.me/seqkit) is a cross-platform and ultrafast toolkit for FASTA/Q file manipulation. For further reading on the `seq` subcommand see the [usage](https://bioinf.shenwei.me/seqkit/usage/#seq).


## Quantification

### pyQUEST

<details markdown="1">
<summary>Output files</summary>

* `pyquest/`
    * `*.query_counts.tsv.gz`: library-independent counts representing the abundance of each unique read sequence.
    * `*.lib_counts.tsv.gz`: library-dependent counts representing the abundance of each oligo template sequence (only generated when user provides a library).
    * `*.stats.json`: JSON-formatted statistics.

</details>

[pyQUEST](https://github.com/cancerit/pyQUEST) is a Python tool for quantifying read and template sequences from SGE experiments. For further reading and documentation see the [pyQUEST output file descriptions](https://github.com/cancerit/pyQUEST#file-formats).

## Quality control


### SeqKit stats

<details markdown="1">
<summary>Output files</summary>

* `seqkit_stats/`
    * `*_[filtered|merged|trimmed|raw].stats.txt`: Basic sequencing statistics.

</details>

[SeqKit](https://bioinf.shenwei.me/seqkit) is a cross-platform and ultrafast toolkit for FASTA/Q file manipulation. For further reading on the `stats` subcommand see the [usage](https://bioinf.shenwei.me/seqkit/usage/#stats).

### FastQC

<details markdown="1">
<summary>Output files</summary>

* `fastqc/`
    * `*_fastqc.html`: FastQC report containing quality metrics.
    * `*_fastqc.zip`: Zip archive containing the FastQC report, tab-delimited data file and plot images.

</details>

[FastQC](http://www.bioinformatics.babraham.ac.uk/projects/fastqc/) gives general quality metrics about your sequenced reads. It provides information about the quality score distribution across your reads, per base sequence content (%A/T/G/C), adapter contamination and overrepresented sequences. For further reading and documentation see the [FastQC help pages](http://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/).

> **NB:** The FastQC plots displayed in the MultiQC report shows _untrimmed_ reads. They may contain adapter sequence and potentially regions with low quality.

### MultiQC

<details markdown="1">
<summary>Output files</summary>

* `multiqc/`
    * `multiqc_report.html`: a standalone HTML file that can be viewed in your web browser.
    * `multiqc_data/`: directory containing parsed statistics from the different tools used in the pipeline.
    * `multiqc_plots/`: directory containing static images from the report in various formats.

</details>

[MultiQC](http://multiqc.info) is a visualization tool that generates a single HTML report summarising all samples in your project. Most of the pipeline QC results are visualised in the report and further statistics are available in the report data directory.

Results generated by MultiQC collate pipeline QC from supported tools e.g. FastQC. The pipeline has special steps which also allow the software versions to be reported in the MultiQC output for future traceability. For more information about how to use MultiQC reports, see <http://multiqc.info>.

## Pipeline information

<details markdown="1">
<summary>Output files</summary>

* `pipeline_info/`
    * Reports generated by Nextflow: `execution_report.html`, `execution_timeline.html`, `execution_trace.txt` and `pipeline_dag.dot`/`pipeline_dag.svg`.
    * Reports generated by the pipeline: `pipeline_report.html`, `pipeline_report.txt` and `software_versions.tsv`.
    * Reformatted samplesheet files used as input to the pipeline: `samplesheet.valid.csv`.

</details>

[Nextflow](https://www.nextflow.io/docs/latest/tracing.html) provides excellent functionality for generating various reports relevant to the running and execution of the pipeline. This will allow you to troubleshoot errors with the running of the pipeline, and also provide you with other information such as launch commands, run times and resource usage.
